presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkAbstractTestQueries.java

```diff
@@ -148,4 +176,71 @@ public void testExplainDdl()
     {
         // DDL statements are not supported by Presto on Spark
     }
+
+    @Test
+    public void testAssignSourceDistributionSplits()
+    {
+        int maxSplitsSize = 2048;
+        Session session = Session.builder(getSession())
+                .setSystemProperty(MAX_SPLITS_DATA_SIZE_PER_SPARK_PARTITION, maxSplitsSize + "B")
+                .build();
+
+        List<ScheduledSplit> splits = new ArrayList<>();
+        long totalSizeInBytes = 0;
+
+        for (int i = 0; i < 1000; ++i) {
+            long splitSizeInBytes = ThreadLocalRandom.current().nextInt((int) (maxSplitsSize * 1.1));
+            totalSizeInBytes += splitSizeInBytes;
+
+            HiveSplit hiveSplit = new HiveSplit(
+                    "test_schema",
+                    "test_table",
+                    "",
+                    "path",
+                    0,
+                    splitSizeInBytes,
+                    splitSizeInBytes,
+                    new Storage(
+                            StorageFormat.create("serde", "input", "output"),
+                            "location",
+                            Optional.empty(),
+                            false,
+                            ImmutableMap.of(),
+                            ImmutableMap.of()),
+                    ImmutableList.of(),
+                    ImmutableList.of(),
+                    OptionalInt.empty(),
+                    OptionalInt.empty(),
+                    NO_PREFERENCE,
+                    1,
+                    ImmutableMap.of(),
+                    Optional.empty(),
+                    false,
+                    Optional.empty(),
+                    NO_CACHE_REQUIREMENT,
+                    Optional.empty());
+            Split testSplit = new Split(new ConnectorId("test"), TestingTransactionHandle.create(), hiveSplit);
+            ScheduledSplit scheduledSplit = new ScheduledSplit(0, new PlanNodeId("source"), testSplit);
+            splits.add(scheduledSplit);
+        }
+
+        SetMultimap<Integer, ScheduledSplit> assignedSplits = assignSourceDistributionSplits(session, splits);
+        asMap(assignedSplits).forEach((partitionId, scheduledSplits) -> {
+            if (scheduledSplits.size() > 1) {
+                long totalPartitionSize = scheduledSplits.stream()
+                        .mapToLong(split -> split.getSplit().getConnectorSplit().getSplitSizeInBytes().getAsLong())
+                        .sum();
+                assertTrue(totalPartitionSize <= maxSplitsSize, format("Total size for splits in one partition should be less than %d", maxSplitsSize));
```

Comment:
Instead of having very generic assertions I would recommend adding more concrete assertions